### PR TITLE
Properly store multiple G4 channel in the parameters.

### PR DESCRIPTION
**Pull request by gbalduzz**
Depends on #138.

- Fixes issue 137 from DCA-develop by removing the duplicated parameter.
- Allow the code to be more maintainable with respect to the addition of new G4 channels.

- The function class now copies the name during construction. This violates the rule that copy/move constructors should behave identically to the assignment operator, but this allows the names a vector of functions to not break upon reallocation, which is particularly important if the code or tests rely on the name. The rationale would be that we do not want to erase a pre-existing name upon copying the values, but we are free to utilize another name while constructing a new function.

The wiki needs to be updated upon approval.

### CODE REVIEW

File: include/dca/phys/dca_data/dca_data.hpp [hunk 1]
```diff
@@ -287,60 +287,11 @@ DcaData<Parameters>::DcaData(Parameters& parameters_ref)
...
-      if (parameters_.accumulateG4ParticleHoleLongitudinalUpUp())
-        G4_err_.emplace_back("G4_" + toString(PARTICLE_HOLE_LONGITUDINAL_UP_UP) + "_err");
-
-      if (parameters_.accumulateG4ParticleHoleLongitudinalUpDown())
-        G4_err_.emplace_back("G4_" + toString(PARTICLE_HOLE_LONGITUDINAL_UP_DOWN) + "_err");
-
-      if (parameters_.accumulateG4ParticleParticleUpDown())
-        G4_err_.emplace_back("G4_" + toString(PARTICLE_PARTICLE_UP_DOWN) + "_err");
-    }
+  for (auto channel : parameters_.get_channels()) {
```
[the remark this answers is not in the document]
Comment (gbalduzz) [reply include/dca/phys/dca_data/dca_data.hpp]: This is part of the analysis code, which does not support multiple channels. And, far from the ideal case, is not that used compared to custom python scripts as it is not really tested for multi bands.
Besides changing the code to analyze multiple channels in a loop I can see two simple solutions:
* Only allow the analysis of a single channel, keeping the same functionality as before.
* select the channel form the JSON input, check that only one is present, and throw an error if said channel is not present in the hdf5 file.

[the remark this answers is not in the document]
Comment (PDoakORNL) [reply include/dca/phys/dca_data/dca_data.hpp]: For now we go with the first option and add an issue to enhance this.

File: include/dca/phys/parameters/four_point_parameters.hpp [hunk 2]
```diff
@@ -58,68 +53,21 @@ class FourPointParameters {
...
+    return channels_;
   }
 
-  bool accumulateG4ParticleHoleLongitudinalUpDown() const {
-    return ph_longitudinal_up_down_;
-  }
-  void accumulateG4ParticleHoleLongitudinalUpDown(const bool accumulate_channel) {
-    ph_longitudinal_up_down_ = accumulate_channel;
+  // Returns true if at least one channel is accumulated.
+  bool accumulateG4() const {
```
[the remark this answers is not in the document]
Comment (gbalduzz) [reply include/dca/phys/parameters/four_point_parameters.hpp]: agreed. Renamed in 858c184a8daf3f82ae2f01d9b17fe1f57133b1bd

File: include/dca/phys/parameters/four_point_parameters.hpp [hunk 3]
```diff
@@ -58,68 +53,21 @@ class FourPointParameters {
...
-  void accumulateG4ParticleHoleCharge(const bool accumulate_channel) {
-    ph_charge_ = accumulate_channel;
-  }
-
-  bool accumulateG4ParticleHoleLongitudinalUpUp() const {
-    return ph_longitudinal_up_up_;
-  }
-  void accumulateG4ParticleHoleLongitudinalUpUp(const bool accumulate_channel) {
-    ph_longitudinal_up_up_ = accumulate_channel;
+  const auto& get_channels() const {
```
[the remark this answers is not in the document]
Comment (gbalduzz) [reply include/dca/phys/parameters/four_point_parameters.hpp]: agreed, renamed in 858c184a8daf3f82ae2f01d9b17fe1f57133b1bd

File: include/dca/phys/dca_data/dca_data.hpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (gbalduzz) [reply include/dca/phys/dca_data/dca_data.hpp]: Actually with aeea48c5825c67d450b848ff5a66cf5b6cf1c7b1 second is already done: the channel is selected by the json file, while the function is read based on its name, so it should be fine.

[the remark this answers is not in the document]
Comment (PDoakORNL) [reply include/dca/phys/dca_data/dca_data.hpp]: This is better

**Comment by uhaehner** [comment]
@gbalduzz Why did you copy the parameters wiki page to the main repo?
The wiki is already a stand-alone git repo.
Now there are two parameters wiki pages to maintain.
Also, neither of them contain the updates corresponding to this PR.
@gbalduzz Please fix this by
- Removing the copy of the wiki in the main repo,
- and updating the *Wiki*.

**Pull request merged**